packages/truffle-codec/lib/types/errors.ts

```diff
@@ -49,10 +49,14 @@ export class NoDefinitionFoundForABIEntryError extends Error {
 //type.
 export class StopDecodingError extends Error {
   public error: Errors.DecoderError;
-  constructor(error: Errors.DecoderError) {
+  public allowRetry: boolean; //setting this to true means that, if the error occurs
+  //when decoding in full mode, we allow an ABI-mode retry.  (if we were already in
+  //ABI mode, we give up.)
+  constructor(error: Errors.DecoderError, allowRetry?: boolean) {
     const message = `Stopping decoding: ${error.kind}`; //sorry about the bare-bones message,
     //but again, users shouldn't actually see this, so I think this should suffice for now
     super(message);
     this.error = error;
+    this.allowRetry = Boolean(allowRetry);
```

Comment:
why capital-B boolean here?

Reply:
Because capital-B `Boolean` is how you convert to a lowercase-b `boolean` in Javascript.  Weird, right?  It's the same with `Number`, `String`, and maybe other things.  You might think that `Boolean` converts to a `Boolean` (which we definitely don't want!), not a `boolean`, but no, that's `new Boolean`.  `Boolean` used without `new` makes a `boolean`.  Again, it's the same with `Number` and `String`, although with `String` I guess you'd usually use `.toString()` instead.

Of course in the case of `Boolean` you can use `!!` instead, but I think it's silly to express as two operations what is really one operation.

Note that in this case I'm just using it to convert `undefined` to `false`, so I could also write it as `allowRetry || false`.  But that seems less direct and less clear to me.

Reply:
Heh oh right I forgot about that